File: ClinicManagementSystem/ClinicManagementSystem.Domain/Entities/Visit.cs
using System;

namespace ClinicManagementSystem.Domain.Entities
{
    public class Visit
    {
        public int Id { get; set; }
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public DateTime VisitDate { get; set; }
        public string Reason { get; set; } = string.Empty;
        public string? Diagnosis { get; set; }
        public string? Status { get; set; } // Matches the frontend "Status" we fixed earlier!

        // Navigation property back to the Patient
        public Patient? Patient { get; set; }
    }
}

File: ClinicManagementSystem.AngularClient1/src/app/services/visit.ts
export interface Visit {
  id?: number;
  patientId: number;
  patientName?: string;
  doctorId?: number;
  doctorName?: string;
  visitDate: string;
  reason?: string;
  diagnosis?: string;
  notes?: string;
  status: 'Pending' | 'Completed';
}